utils/misc.ml

```diff
@@ -2023,11 +2027,25 @@ module Maybe_bounded = struct
     | Bounded r when r.bound > 0 -> r.bound <- r.bound - 1
     | Bounded _ -> ()
 
+  let incr = function
+    | Unbounded -> ()
+    | Bounded r -> r.bound <- r.bound + 1
+
   let is_depleted = function
     | Unbounded -> false
     | Bounded r -> r.bound <= 0
 
+  let is_in_bounds n = function
```

Comment:
These two functions should account for `n` being provided as negative, probably?